contracts/serverless-v2/CommonChainJobs.sol

```diff
@@ -315,7 +317,10 @@ contract CommonChainJobs is
 
         uint256 len = selectedExecutors[key].length;
         for (uint256 index = 0; index < len; index++) {
-            executors.updateOnExecutionTimeoutSlash(selectedExecutors[key][index]);
+            address executorKey = selectedExecutors[key][index];
+            require(!hasExecutedJob[key][executorKey], "JOB_ALREADY_EXECUTED");
+            delete hasExecutedJob[key][executorKey];
+            executors.updateOnExecutionTimeoutSlash(executorKey);
         }
 
```

Comment:
you may delete the `selectedExecutors[key]` after the for loop.

Reply:
it's already there